compiler/rustc_infer/src/infer/error_reporting/suggest.rs

```diff
@@ -461,6 +461,64 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             }
         }
     }
+
+    /// For "one type is more general than the other" errors on closures, suggest changing the lifetime
+    /// of the parameters to accept all lifetimes.
+    pub(super) fn suggest_for_all_lifetime_closure(
+        &self,
+        span: Span,
+        exp_found: &ty::error::ExpectedFound<ty::PolyTraitRef<'tcx>>,
+        diag: &mut Diagnostic,
+    ) {
+        // 1. Get the substs of the closure.
+        // 2. Assume exp_found is FnOnce / FnMut / Fn, we can extract function parameters from [1].
+        let expected = exp_found.expected.try_map_bound(|x| x.substs.get(1).cloned().ok_or(()));
```

Comment:
Can you use `map_bound` and [`transpose`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/sty/struct.Binder.html#method.transpose) instead of `try_map_bound` and `ok_or`?

```suggestion
        let expected = exp_found.expected.map_bound(|x| x.substs.get(1)).transpose();
```

Reply:
Fixed, thanks for the suggestion!